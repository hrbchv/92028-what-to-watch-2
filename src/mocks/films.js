const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Fantasy`,
    year: `2014`
  },
  {
    name: `Bohemian Rhapsody`,
    imgSrc: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    year: `2015`
  },
  {
    name: `Macbeth`,
    imgSrc: `img/macbeth.jpg`,
    genre: `Drama`,
    year: `2016`
  },
  {
    name: `Aviator`,
    imgSrc: `img/aviator.jpg`,
    genre: `Drama`,
    year: `2017`
  },
  {
    name: `Some name`,
    imgSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    year: `2018`
  }
];

export default films;
