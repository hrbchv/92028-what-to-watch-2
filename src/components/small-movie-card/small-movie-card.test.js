import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card.jsx";
import films from "../../mocks/films";

describe(`SmallMovieCard`, () => {
  it(`should renders correctly`, () => {
    const index = 2;
    const activeMovieIndex = 1;
    const movie = films[index];

    const tree = renderer
      .create(<SmallMovieCard
        movie={movie}
        index={index}
        activeMovieIndex={activeMovieIndex}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
