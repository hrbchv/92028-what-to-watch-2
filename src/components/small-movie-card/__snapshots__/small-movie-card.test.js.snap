// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard should renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/macbeth.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Macbeth
    </a>
  </h3>
</article>
`;
